parts/tools/gen_diffuser_frame.py

```
@@ -0,0 +1,763 @@
+            # the two outermost channels are the frame's backbone — never thin
+            if OUTER_RAILS_FULL and s['chan'] in (0, nrails - 1):
+                return False
+            # inner channels: keep 1 in INNER_KEEP_EVERY, staggered per channel,
+            # so consecutive gaps may both be empty
+            return (s['chan'] + s['gap']) % INNER_KEEP_EVERY != 0
+
+        cand = [k for k, s in enumerate(segs)
+                if s['kind'] == 'rail' and seg_len(s) > SHORT_RAIL and droppable(s)]
+        dropped = []
+        for k in cand:
+            trial = [s for idx, s in enumerate(segs) if idx not in dropped + [k]]
```

Ruff (0.16.0)

[warning] Consider `[*dropped, k]` instead of concatenation

Replace with `[*dropped, k]`

(RUF005)